### ISSUE TITLE
Option to automatically collapse arrays and auto format

### ISSUE THREAD

**Issue by ZetaDaemon** [BLCM/OpenBLCMM#37]
Basically as the title says, 99% of the time I want the arrays to be collapsed and then auto formatted.
Since collapse arrays is a toggle, maybe the option could be to set its starting state when making a new tab.
For maybe add a button next to it to toggle "automatic auto formatting" so that it formats objects on displaying them.

**Comment by apocalyptech** [comment]
Yep, agreed!  Getting collapse-arrays to just remember its state should be pretty trivial.  I assume I could figure out a solution for auto-formatting, too

**Comment by apocalyptech** [comment]
Okay, so this is in place in `next`, though I wanted to leave this open until I get some feedback on the implementation.  Rather than having a separate toggle, I ended up just making the button itself a toggle.  There are a couple of cases in which users would end up having to hit some extra buttons -- for instance, if you deformat a dump and then want to go back to fully-formatted, you'd either have to hit "Auto format" *twice* (to turn it off and on again), or spin Deformat to `0` and hit Deformat again.  My current opinion is that this is better than the original state, though, and an overall improvement (and also that I'd prefer it to having two format buttons: one toggleable and one not).

So yeah, will leave this for a bit, but should be in place for now!
